### PR TITLE
HPCC-16661 Added libhiredis submodule

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner <michael.gardner@lexisnexis.com>

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@xwang2713 @richardkchapman please see https://track.hpccsystems.com/browse/HPCC-16661 for discussion

**Review by xwang2713 — COMMENTED** [review]
Looks good

**Comment by Michael-Gardner** [comment]
I've tested this build on centos 6 (no native hiredis package installed) and ubuntu 16.10 (has native hiredis package installed.)  Links correctly to our supplied version.

File: .gitmodules [hunk 1]
```diff
@@ -43,3 +43,6 @@
 [submodule "libcouchbase"]
 	path = plugins/couchbase/libcouchbase
 	url = https://github.com/hpcc-systems/libcouchbase.git
+[submodule "plugins/redis/hiredis"]
+	path = plugins/redis/hiredis
+	url = https://github.com/redis/hiredis.git
```
Comment (richardkchapman) [inline .gitmodules]: As with the other submodules, we need to clone this to ensure that it continues to exist

Comment (richardkchapman) [inline .gitmodules]: Fork done

**Comment by Michael-Gardner** [comment]
@richardkchapman Just updated the URL.  Going to do a quick test build to make sure all is well.

**Comment by Michael-Gardner** [comment]
@richardkchapman tested on centos 6 and ubuntu 16.04.  Worked as before (not that I was expecting anything different, the commit it points to is the same despite the fork).  I went ahead and squashed the commits.

**Pull request merged**